packages/big-design/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,119 @@
+import { ArrowDropDownIcon, ChevronLeftIcon, ChevronRightIcon } from '@bigcommerce/big-design-icons';
+import React, { useEffect, useState } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton } from './styled';
+
+export interface PaginationProps extends MarginProps {
+  currentPage: number;
+  itemsPerPage: number;
+  itemsPerPageOptions: number[];
+  totalItems: number;
+  onPageChange(page: number): void;
+  onItemsPerPageChange(range: number): void;
+}
+
+export const Pagination: React.FC<PaginationProps> = props => {
+  const { itemsPerPage, currentPage, totalItems, itemsPerPageOptions, onPageChange, onItemsPerPageChange } = props;
+
+  const [maxPages, setMaxPages] = useState(Math.ceil(totalItems / itemsPerPage));
+  const [itemRange, setItemRange] = useState({ start: 0, end: 0 });
+
+  const handlePageOutOfBounds = () => {
+    if (currentPage < 1 || isNaN(currentPage) || currentPage === undefined) {
```

Comment:
I'm wondering if we need to verify for `isNaN` or `undefined` when typescript has `currentPage` as `number`. I guess it makes sense to verify for those not using `ts`, not sure what others think?